react/components/SearchQuery.js

```diff
@@ -188,14 +188,22 @@ const useQueries = (
     },
   })
 
-  const { data: { sponsoredProducts } = [] } = useQuery(
+  const { data: { sponsoredProducts } = [], loading: sponsoredProductLoading } = useQuery(
     sponsoredProductsQuery,
     {
       variables,
       skip: sponsoredProductsBehavior === 'skip',
     }
   )
 
+  const sponsoredProductReturn = () => {
```

Comment:
I don't really like this function. Is not very common to do this in JavaScript, specially with it not receiving any params to justify its existence.

Also, you should also check for errors, not only the loading state.

I'd rather you'd just:

```javascript
const showSponsoredProducts = !sponsoredProductsLoading && !sponsoredProductsError
const sponsoredProductsResult = showSponsoredProducts ? sponsoredProducts : []
```

If you feel this is a level of abstraction below the current `useQueries` function (which I do), you can isolate this logic of requesting sponsored products into a different function.

Reply:
Ok, I didn't like the `! && !`, neither the ternary `if`, but as you said, seems to be a prevalent thing in JS.